kite-data/kite-data-crunch/src/main/java/org/kitesdk/data/crunch/DatasetTarget.java

```diff
@@ -99,12 +105,12 @@ public boolean handleExisting(WriteMode writeMode, long lastModForSource,
           LOG.info("Writing to existing dataset/view: " + view);
           break;
         case CHECKPOINT:
-          boolean ready = true; // dataset.isReady(); // TODO: add when CDK-451 is ready
           long lastModForTarget = -1;
           if (view instanceof LastModifiedAccessor) {
             lastModForTarget = ((LastModifiedAccessor) view).getLastModified();
           }
-          if (ready && lastModForTarget > lastModForSource) {
+
+          if (ready && (lastModForTarget > lastModForSource)) {
```

Comment:
With HBase (right now anyway) the behavior would remain unchanged since it doesn't implement LastModifiedAccessor. So an HBase target is always considered to be older than any type of source and will never be reused.

Right now the implementations for Signalable and LastModifiedAccessor overlap so there aren't cases where we have one without the other. If a future Dataset implementation picked up LastModifiedAccsssor in order to use checkpoint writemode safely it seems like it would need to be fully atomic. In which case maybe Signalable isn't all that valuable from a map reduce perspective, since any view would be "ready" its nature.

I'm open to changing it, but it seems like without an "atomic" check that we could assert in code relying only on last modified feels dangerous.